modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/api/ast/ArtifactAstUtils.java

```diff
@@ -107,43 +147,106 @@ public static ArtifactAst parseAndBuildAppExtensionModel(String[] configResource
+   * {@link ExtensionModel} which represents it.
+   *
+   * @param ast                 the artifact's AST
+   * @param artifactClassLoader the artifact's classloader
+   * @param artifactId          the artifact's ID.
+   * @param artifactVersion     the artifact's version.
+   * @param dependencies        the dependencies in context.
+   * @return an optional {@link ExtensionModel}
+   */
+  public static Optional<ExtensionModel> parseArtifactExtensionModel(ArtifactAst ast,
+                                                                     ClassLoader artifactClassLoader,
+                                                                     String artifactId,
+                                                                     Optional<String> artifactVersion,
+                                                                     Set<ExtensionModel> dependencies) {
+    if (!artifactVersion.isPresent()) {
+      logModelNotGenerated("No version specified", artifactId, ast.getArtifactType());
       return empty();
     }
 
-    Optional<ExtensionModelLoader> loader = getOptionalLoaderById(ArtifactAstUtils.class.getClassLoader(), MULE_SDK_LOADER_ID);
+    Optional<ExtensionModelLoader> loader = getOptionalLoaderById(ArtifactAstUtils.class.getClassLoader(),
+                                                                  MULE_SDK_THIS_LOADER_ID);
     if (loader.isPresent()) {
-      Set<ExtensionModel> dependenciesExtensionModels = muleContext.getExtensionManager().getExtensions();
       return of(loader.get()
-          .loadExtensionModel(builder(artifactClassLoader, getDefault(dependenciesExtensionModels))
-              .addParameter(VERSION_PROPERTY_NAME, artifactCoordinates.get().getVersion())
+          .loadExtensionModel(builder(artifactClassLoader, getDefault(dependencies))
+              .addParameter(VERSION_PROPERTY_NAME, artifactVersion.get())
               .addParameter(MULE_SDK_ARTIFACT_AST_PROPERTY_NAME, ast)
-              .addParameter(MULE_SDK_EXTENSION_NAME_PROPERTY_NAME, muleContext.getConfiguration().getId())
-              .addParameter(MULE_SDK_TYPE_LOADER_PROPERTY_NAME, new ApplicationTypeLoader(dependenciesExtensionModels))
+              .addParameter(MULE_SDK_EXTENSION_NAME_PROPERTY_NAME, artifactId)
               .build()));
     } else {
-      logModelNotGenerated("Mule ExtensionModelLoader not found", muleContext);
+      logModelNotGenerated("Mule ExtensionModelLoader not found", artifactId, ast.getArtifactType());
       return empty();
     }
   }
 
+  /**
+   * Handles a {@link ValidationResult} uniformly. Warnings are logged, errors raise exceptions.
+   *
+   * @param validationResult The {@link ValidationResult} obtained using an {@link ArtifactAstValidator}.
+   * @throws ConfigurationException If there was any result item with {@link ERROR} level.
+   */
+  public static void handleValidationResult(ValidationResult validationResult)
```

Comment:
You're putting this method as API but is only used once in an internal class. This should simply be a private method of the factory in which you use it, or at most, part of an internal utils. Don't make things API unnecesarily.

Reply:
You are right. I just moved it to its own utility class on an internal package. Thanks